### PR TITLE
gh-123275: Support `-Xgil=1` and `PYTHON_GIL=1` on non-free-threaded builds

### PR DESCRIPTION
**Pull request by ZeroIntensity**
cc @godlygeek

* Issue: gh-123275

### CODE REVIEW

File: Misc/NEWS.d/next/Core_and_Builtins/2024-08-23-18-31-10.gh-issue-123275.DprIrj.rst [hunk 1]
```diff
@@ -0,0 +1 @@
+Support ``-Xgil=1`` and ``PYTHON_GIL=1`` on non-free-threaded builds.
```
Comment (Eclips4) [inline Misc/NEWS.d/next/Core_and_Builtins/2024-08-23-18-31-10.gh-issue-123275.DprIrj.rst]:
```suggestion
Support :option:`-X gil=1 <-X>` and :envvar:`PYTHON_GIL=1 <PYTHON_GIL>` on non-free-threaded builds.
```

File: Doc/using/cmdline.rst [hunk 2]
```diff
@@ -1221,13 +1221,12 @@ conflict.
 .. envvar:: PYTHON_GIL
 
    If this variable is set to ``1``, the global interpreter lock (GIL) will be
-   forced on. Setting it to ``0`` forces the GIL off.
+   forced on. Setting it to ``0`` forces the GIL off (needs Python configured with
+   the :option:`--disable-gil` build option.)
```
Comment (godlygeek) [inline Doc/using/cmdline.rst]: Minor nitpick:
```suggestion
   the :option:`--disable-gil` build option).
```

**Review by colesbury — APPROVED** [review]
I don't have a strong opinion about this one way or the other. This seems fine to me.

**Comment by ZeroIntensity** [comment]
I think this needs the 3.13 backport label, unless this is considered to be a feature.

**Review by corona10 — APPROVED** [review]
lgtm

**Comment by corona10** [comment]
@ZeroIntensity I think that we forgot to update test code.
https://github.com/python/cpython/blob/b5aa271f86229f126c21805ff2bd3b95526818a4/Lib/test/test_cmd_line.py#L885-L914

Can you update it too?

**Comment by ZeroIntensity** [comment]
Sure, I'll submit a hotfix. Sorry about that!

**Comment by corona10** [comment]
> Sure, I'll submit a hotfix. Sorry about that!

No problem, it was my fault since I approved and merged this PR without checking test code.
Thanks for the work!

**Comment by ZeroIntensity** [comment]
FWIW, that test doesn't actually run on default builds. I'll just add a new test next to it to make sure that it works.

**Comment by corona10** [comment]
> FWIW, that test doesn't actually run on default builds. I'll just add a new test next to it to make sure that it works.

Yeah, I know, but we should add the test when we want to support this feature at the default build, that was what I wanted to mean :)

**Pull request merged**